src/app/common/hooks/account/use-account-names.ts

```diff
@@ -23,12 +21,15 @@ export function useCurrentAccountDisplayName() {
 }
 
 export function useAccountDisplayName({ address, index }: { index: number; address: string }) {
-  const { data: names = [], isLoading } = useGetAccountNamesByAddressQuery(address);
-  return useMemo(() => {
-    const name = names[0] || getAutogeneratedAccountDisplayName(index);
-    return {
-      name,
-      isLoading,
-    };
-  }, [names, index, isLoading]);
+  const query = useGetBnsNamesOwnedByAddress(address, {
+    select: resp => {
+      const names = resp.names ?? [];
+      return names[0] || getAutogeneratedAccountDisplayName(index);
+    },
+  });
+
+  return {
+    ...query,
+    data: query.data || getAutogeneratedAccountDisplayName(index),
+  };
```

Comment:
We only need one of these two techniques right? select or replacing data?

Reply:
problem here is again in ledger mode, when query is not enabled, select doesn't work

Reply:
but if you do it on the data prop, it works for both cases, and the select isn't needed?

Reply:
you mean 
    `data: (query.data.names ?? [])[0] || getAutogeneratedAccountDisplayName(index)` ?

Reply:
you can try it in ledger bitcoin only mode